src/Symfony/Component/VarDumper/VarDumper.php

```diff
@@ -37,14 +42,25 @@ class VarDumper
      */
     private static $handler;
 
+    private static ?string $prevOptionsHash = null;
+
+    private static bool $manualHandlerRegister = false;
+
     /**
      * @param string|null $label
+     * @param VarDumperOptions|array $options
      */
-    public static function dump(mixed $var/* , string $label = null */)
+    public static function dump(mixed $var/* , string $label = null, VarDumperOptions|array $options = [] */)
     {
         $label = 2 <= \func_num_args() ? func_get_arg(1) : null;
-        if (null === self::$handler) {
-            self::register();
+        $options = 3 <= \func_num_args() ? func_get_arg(2) : [];
+
+        if (!$options instanceof VarDumperOptions) {
+            $options = new VarDumperOptions($options);
+        }
+
+        if (self::requiresRegister($options)) {
+            self::register($options);
```

Comment:
this will change the options of the default handler when any options changes?
this would be unexpected to me.

Reply:
`requiresRegister` helps on taking care of registering again the handler if `dump/dd` is not called with the exact same set of options as the previous call (this should work great with options defined in an env var!). So if you call the default dumper (I assume what you called "default dumper" is the one called without any special option? I may be wrong 😄), it should be registered again with its default options.

I struggled a bit on this part, only a few options need a full and new registration. I may be missing something?